### PR TITLE
Update README.md Install Instructions

### PR DESCRIPTION
**Pull request by Beagle123**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,38 +5,88 @@
 Ruby bindings for GNOME
 
 This is a set of bindings for the GNOME 3.x and 4.x libraries to use
-from Ruby.
+from Ruby that allow you to write graphical user interaces in ruby.
```
Comment (kou) [inline README.md:8]:
```suggestion
from Ruby that allows you to write graphical user interfaces in Ruby.
```

File: README.md [hunk 2]
```diff
@@ -5,38 +5,88 @@
...
 
-## Installation
+## Installation Instructions for gtk3 and gtk4
 
-### Install the current releases
+### Debian/Ubuntu
 
-The gems of the Ruby-GNOME projects can be easily installed with the `gem` command.
+Ruby can be run from the system directories, or from your home directory, or both.  
+Most people won't have the correct permissions for a system intsall, So the vast majority of
```
Comment (kou) [inline README.md:15]:
```suggestion
Most people won't have the correct permissions for a system install, so the vast majority of
```

File: README.md [hunk 3]
```diff
@@ -5,38 +5,88 @@
...
+people to run ruby from their local, <b>home/...</b> folder.
```
Comment (kou) [inline README.md:16]:
```suggestion
people to run Ruby from their local, `/home/...` directory.  
```

File: README.md [hunk 4]
```diff
@@ -5,38 +5,88 @@
...
-### Install the current releases
+### Debian/Ubuntu
 
-The gems of the Ruby-GNOME projects can be easily installed with the `gem` command.
+Ruby can be run from the system directories, or from your home directory, or both.  
+Most people won't have the correct permissions for a system intsall, So the vast majority of
+people to run ruby from their local, <b>home/...</b> folder.  
 
-Here is an example to install the gtk4 gem and its dependencies:
+To test if your copy of ruby is running locally enter this on the command line:
```
Comment (kou) [inline README.md:18]:
```suggestion
To test if your copy of Ruby is running locally enter this on the command line:
```

File: README.md [hunk 5]
```diff
@@ -5,38 +5,88 @@
...
+
+```bash
+which ruby
+```
+
+If it outputs something like this:
+```bash
+/home/ralph/.rbenv/shims/ruby
+```
+that starts with "home", that's local (good!)
```
Comment (kou) [inline README.md:28]:
```suggestion
that starts with `/home/`, that's local (good!) 
```

File: README.md [hunk 6]
```diff
@@ -5,38 +5,88 @@
...
+```
+that starts with "home", that's local (good!) 
+
+But if outputs something like this:
+```bash
+/usr/bin/ruby
+```
+That is a system folder, and the gtk3 gem will fail to install.  Actually,
+a system administrator with super-user permissions could install everything
+in the system folders (see advanced install), but its much better for everyone else to run ruby locally.
```
Comment (kou) [inline README.md:36]:
```suggestion
in the system folders (see advanced install), but its much better for everyone else to run Ruby locally.
```

File: README.md [hunk 7]
```diff
@@ -5,38 +5,88 @@
...
+
+If you're already running a copy of ruby under your home folder, you can install gtk4 (or gtk3):
```
Comment (kou) [inline README.md:38]:
```suggestion
If you're already running a copy of Ruby under your home directory, you can install gtk4 (or gtk3):
```

File: README.md [hunk 8]
```diff
@@ -5,38 +5,88 @@
...
 
 ```bash
 gem install gtk4
 ```
 
-Here are instructions for individual environments.
+But if you're running ruby from the system folders, you can install a ruby version manager
```
Comment (kou) [inline README.md:44]:
```suggestion
But if you're running ruby from the system folders, you can install a Ruby version manager
```

File: README.md [hunk 9]
```diff
@@ -5,38 +5,88 @@
...
+like <a href=https://rvm.io>RVM</a> or <a href="https://github.com/rbenv/rbenv?tab=readme-ov-file">rbenv</a>.
```
Comment (kou) [inline README.md:45]:
```suggestion
like [RVM](https://rvm.io/) or [rbenv](https://github.com/rbenv/rbenv).  
```

File: README.md [hunk 10]
```diff
@@ -5,38 +5,88 @@
...
+If you're already running a copy of ruby under your home folder, you can install gtk4 (or gtk3):
 
 ```bash
 gem install gtk4
 ```
 
-Here are instructions for individual environments.
+But if you're running ruby from the system folders, you can install a ruby version manager
+like <a href=https://rvm.io>RVM</a> or <a href="https://github.com/rbenv/rbenv?tab=readme-ov-file">rbenv</a>.  
+These programs allow you to run multiple versions of ruby from your home directory.
```
Comment (kou) [inline README.md:46]:
```suggestion
These programs allow you to run multiple versions of Ruby from your home directory.
```

File: README.md [hunk 11]
```diff
@@ -5,38 +5,88 @@
...
+
+rbenv has a <a href=https://github.com/rbenv/rbenv-installer#rbenv-installer>rvenv-installer</a>
```
Comment (kou) [inline README.md:48]:
```suggestion
rbenv has a [rbenv-installer](https://github.com/rbenv/rbenv-installer)
```

File: README.md [hunk 12]
```diff
@@ -5,38 +5,88 @@
...
+that installs everything and adds a line to your .bashrc file so you're running
```
Comment (kou) [inline README.md:49]:
```suggestion
that installs everything and adds a line to your `.bashrc` file so you're running
```

File: README.md [hunk 13]
```diff
@@ -5,38 +5,88 @@
...
+local copies of ruby.
```
Comment (kou) [inline README.md:50]:
```suggestion
local copies of Ruby.
```

File: README.md [hunk 14]
```diff
@@ -5,38 +5,88 @@
...
+```bash
+/home/ralph/.rbenv/shims/ruby
+```
+that starts with "home", that's local (good!) 
+
+But if outputs something like this:
+```bash
+/usr/bin/ruby
+```
+That is a system folder, and the gtk3 gem will fail to install.  Actually,
```
Comment (kou) [inline README.md:34]:
```suggestion
That is a system directory, and the gtk4 gem will fail to install.  Actually,
```

File: README.md [hunk 15]
```diff
@@ -5,38 +5,88 @@
...
+like <a href=https://rvm.io>RVM</a> or <a href="https://github.com/rbenv/rbenv?tab=readme-ov-file">rbenv</a>.  
+These programs allow you to run multiple versions of ruby from your home directory.
+
+rbenv has a <a href=https://github.com/rbenv/rbenv-installer#rbenv-installer>rvenv-installer</a>
+that installs everything and adds a line to your .bashrc file so you're running
+local copies of ruby.
 
-#### Debian/Ubuntu
+To get rvenv:
+1) Run the installer
```
Comment (kou) [inline README.md:53]:
```suggestion
1. Run the installer
```

File: README.md [hunk 16]
```diff
@@ -5,38 +5,88 @@
...
+2) Restart your command prompt
```
Comment (kou) [inline README.md:54]:
```suggestion
2. Restart your command prompt
```

File: README.md [hunk 17]
```diff
@@ -5,38 +5,88 @@
...
+3) Run $rbenv install --list    #(find recent ruby version)
```
Comment (kou) [inline README.md:55]:
```suggestion
3. Run `rbenv install --list` (find recent Ruby version)
```

File: README.md [hunk 18]
```diff
@@ -5,38 +5,88 @@
...
+4) Run $rbenv install 3.3.5      #(example version.  choose one from list)
```
Comment (kou) [inline README.md:56]:
```suggestion
4. Run `rbenv install 3.3.5` (example version.  choose one from list)
```

File: README.md [hunk 19]
```diff
@@ -5,38 +5,88 @@
...
+5) Run $which ruby              #(check that its running from home directory
```
Comment (kou) [inline README.md:57]:
```suggestion
5. Run `which ruby` (check that its running from home directory)
```

File: README.md [hunk 20]
```diff
@@ -5,38 +5,88 @@
...
-#### macOS
+### macOS
+
+Input needed...
+
+### Windows
+
+In windows, the gem should install without issue.  If you install ruby using the <a href="https://rubyinstaller.org/">
+Ruby Installer</a>, make sure that MSYS2 gets installed too.  Gtk needs MSYS2 to make the binary filees.
+Then, at the ruby command prompt:
```
Comment (kou) [inline README.md:73]:
```suggestion
In Windows, the gem should be installed without issue.  If you install Ruby using the
[Ruby Installer](https://rubyinstaller.org/), make sure that MSYS2 gets installed too.  GTK needs MSYS2 to make the binary files.
Then, at the Ruby command prompt:
```

File: README.md [hunk 21]
```diff
@@ -5,38 +5,88 @@
...
 
-#### Windows
 
-...
+## Advanced Installation instructions
+
+### Debian/Ubuntu
+
+```bash
+apt install -y gcc make ruby-dev
```
Comment (kou) [inline README.md:85]:
```suggestion
sudo apt install -y gcc make ruby-dev
```

**Comment by Beagle123** [comment]
That looks great.  I think it will help a lot of people like me who don't know why the gtk gem ins't installing.

**Comment by kou** [comment]
Thanks.

**Pull request merged**